### DISCUSSION TITLE
Add subblocks as references

### DISCUSSION THREAD

**Discussion by olofk** [FPGAwars/icestudio#832] [Ideas]
When I add a block to a project, icestudio seems to make an internal copy of that block, so when I update the original block, it is not changed in the new project. This makes it hard to reuse components. It would be very useful to add the block as a reference to another .ice file instead so that it's updated when there are changes to the subblock

**Comment by drtrigon** [comment]
When placing (or may be when saving) the user should be able to choose between:
- place copy
- place ref
As booth behaviours have their advantages. I would not like to open a project after some months again in order to find it does not work anymore just because some other files have changed.

Am 21. November 2019 23:33:46 MEZ schrieb Olof Kindgren <notifications@github.com>:

**Comment by cavearr** [comment]
At present icestudio not support block reference but i'm labelling this issue as new feature becasuse it is a very interesting feature for next versions.

Thanks for your comments!

**Comment by cavearr** [comment]
Hello! I am reorganizing the issues, trying to give more visibility and be able to better organize and prioritize the work and my roadmap (I will publish it very soon), I am converting some issues that are **new feature requests** as **ideas** on Github **discussions**.

Thanks for understanding.
